### PR TITLE
Selim Osmanow HomeWork#1

### PR DESCRIPTION
**Pull request by theDDk**

### CODE REVIEW

File: Lesson1/Selim_Osmanov/HomeTask1.1/Makefile.cmake [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# CMAKE generated file: DO NOT EDIT!
```
Comment (smay1613) [inline Lesson1/Selim_Osmanov/HomeTask1.1/Makefile.cmake:1]: why it is in pull request and in your repo?

File: Lesson1/Selim_Osmanov/HomeTask1.1/1 [hunk 2]
```diff
@@ -0,0 +1 @@
+
```
Comment (smay1613) [inline Lesson1/Selim_Osmanov/HomeTask1.1/1:1]: _remove this file_

File: Lesson1/Selim_Osmanov/HomeTask1.1/ht1_1head.h [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#pragma once
```
Comment (smay1613) [inline Lesson1/Selim_Osmanov/HomeTask1.1/ht1_1head.h:1]: please use more descriptive names for files

File: Lesson1/Selim_Osmanov/HomeTask1.1/ht1_1head.h [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+#include <vector>
+#include <iostream>
+
+void EnterElements(std::vector<int>& FirstVector, size_t SizeOfFirstVector);
+void ShowElements(std::vector<int> FirstVector, size_t SizeOfFirstVector);
+void UniteVectors(std::vector<int>& LastVector, std::vector<int> FirstVector,
+                  std::vector<int> SecondVector, size_t SizeOfFirstVector,
```
Comment (smay1613) [inline Lesson1/Selim_Osmanov/HomeTask1.1/ht1_1head.h:8]: That's awful. For what purpose there are size of vectors?

File: Lesson1/Selim_Osmanov/HomeTask1.1/ht1_1head.h [hunk 5]
```diff
@@ -0,0 +1,9 @@
+#pragma once
+#include <vector>
+#include <iostream>
+
+void EnterElements(std::vector<int>& FirstVector, size_t SizeOfFirstVector);
+void ShowElements(std::vector<int> FirstVector, size_t SizeOfFirstVector);
+void UniteVectors(std::vector<int>& LastVector, std::vector<int> FirstVector,
```
Comment (smay1613) [inline Lesson1/Selim_Osmanov/HomeTask1.1/ht1_1head.h:7]: What the difference between LastVector/First/SecondVector? Names are bad.

File: Lesson1/Selim_Osmanov/HomeTask1.1/ht1_1head.h [hunk 6]
```diff
@@ -0,0 +1,9 @@
+#pragma once
+#include <vector>
+#include <iostream>
+
+void EnterElements(std::vector<int>& FirstVector, size_t SizeOfFirstVector);
+void ShowElements(std::vector<int> FirstVector, size_t SizeOfFirstVector);
```
Comment (smay1613) [inline Lesson1/Selim_Osmanov/HomeTask1.1/ht1_1head.h:6]: Why are you passing container by copy?

File: Lesson1/Selim_Osmanov/HomeTask1.1/ht1_1lib.cpp [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+void ShowElements(std::vector<int> FirstVector, size_t SizeOfFirstVector){
+    for (size_t i = 0; i < SizeOfFirstVector; ++i){
+        std::cout << FirstVector[i] << " ";
+    }
+}
+
+void UniteVectors(std::vector<int>& LastVector, std::vector<int> FirstVector, std::vector<int> SecondVector, size_t SizeOfFirstVector, size_t SizeOfLastVector){
+    for (size_t i = 0; i < SizeOfFirstVector; ++i){
```
Comment (smay1613) [inline Lesson1/Selim_Osmanov/HomeTask1.1/ht1_1lib.cpp:19]: **You're trying to manipulate vectors in C-way.**
Don't do that.

File: Lesson1/Selim_Osmanov/HomeTask1.1/ht1_1lib.cpp [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+    }
+}
+
+void ShowElements(std::vector<int> FirstVector, size_t SizeOfFirstVector){
+    for (size_t i = 0; i < SizeOfFirstVector; ++i){
+        std::cout << FirstVector[i] << " ";
+    }
+}
+
+void UniteVectors(std::vector<int>& LastVector, std::vector<int> FirstVector, std::vector<int> SecondVector, size_t SizeOfFirstVector, size_t SizeOfLastVector){
```
Comment (smay1613) [inline Lesson1/Selim_Osmanov/HomeTask1.1/ht1_1lib.cpp:18]: _vector can get it's size with .size() method_

File: Lesson1/Selim_Osmanov/HomeTask1.2/ht1_2lib.cpp [hunk 9]
```diff
@@ -0,0 +1,21 @@
+#include <iostream>
+#include <list>
+#include <cmath>
+void TakeNumbers(size_t Exponent, size_t Basis, std::list<size_t>& OutList){
```
Comment (smay1613) [inline Lesson1/Selim_Osmanov/HomeTask1.2/ht1_2lib.cpp:4]: This function must return not the std::list, but some integer value.
std::list just should help you to solve the task.

File: Lesson1/Selim_Osmanov/HomeTask1.2/ht1_2lib.cpp [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+            firstNumber /= 10;
+        };
+        size_t lastNumber = var%10;
+        OutList.push_back(firstNumber);
+        OutList.push_back(lastNumber);
+    }
+}
+
+void ShowResult(std::list<size_t>& OutList){
+    for (std::list<size_t>::iterator i = OutList.begin(); i != OutList.end(); ++i){
```
Comment (smay1613) [inline Lesson1/Selim_Osmanov/HomeTask1.2/ht1_2lib.cpp:18]: _Range for_

File: Lesson1/Selim_Osmanov/HomeTask1.2/ht1_2lib.cpp [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+        while(firstNumber >= 10){
+            firstNumber /= 10;
+        };
+        size_t lastNumber = var%10;
+        OutList.push_back(firstNumber);
+        OutList.push_back(lastNumber);
+    }
+}
+
+void ShowResult(std::list<size_t>& OutList){
```
Comment (smay1613) [inline Lesson1/Selim_Osmanov/HomeTask1.2/ht1_2lib.cpp:17]: _Why not const?_

File: Lesson1/Selim_Osmanov/HomeTask1.3/ht1_3lib.cpp [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+}
+
+void ShowElements (std::forward_list<int>& MyForwardList) {
+    for(std::forward_list<int>::iterator i = MyForwardList.begin(); i != MyForwardList.end(); ++i){
+        std::cout << *i << " ";
+    }
+}
+
+void ReverseElements (std::forward_list<int>& MyForwardList, std::forward_list<int>& MyNewForwardList) {
+    for(std::forward_list<int>::iterator i = MyForwardList.begin(); i != MyForwardList.end(); ++i) {
```
Comment (smay1613) [inline Lesson1/Selim_Osmanov/HomeTask1.3/ht1_3lib.cpp:18]: **This function should modify the input container**

File: Lesson1/Selim_Osmanov/HomeTask1.2/1 [hunk 13]
```diff
@@ -0,0 +1 @@
+
```
Comment (smay1613) [inline Lesson1/Selim_Osmanov/HomeTask1.2/1:1]: Copy-paste is not a good practice.

**Review by smay1613 — CHANGES_REQUESTED** [review]

File: Lesson1/Selim_Osmanov/HomeTask1.2/1 (hunk 13 above)
Comment (theDDk) [reply Lesson1/Selim_Osmanov/HomeTask1.2/1:1]: It is not copy-past. It was written by me.

Comment (smay1613) [reply Lesson1/Selim_Osmanov/HomeTask1.2/1:1]: You've copy-pasted directory, so I see duplicated empty file "1"

**Pull request closed without merging**